src/mesh_analysis.rs

```diff
@@ -653,4 +720,79 @@ mod tests {
         );
         assert_eq!(genus, 3);
     }
+
+    #[test]
+    fn test_border_edge_loops_should_be_one() {
+        let (faces, vertices) = tessellated_triangle();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces.clone(),
+            vertices.clone(),
+            NormalStrategy::Sharp,
+        );
+
+        let oriented_edges: Vec<OrientedEdge> = geometry.oriented_edges_iter().collect();
+        let edge_sharing_map = edge_sharing(&oriented_edges);
+
+        let correct_loop: Vec<UnorientedEdge> = vec![
+            UnorientedEdge(OrientedEdge::new(0, 1)),
+            UnorientedEdge(OrientedEdge::new(1, 2)),
+            UnorientedEdge(OrientedEdge::new(2, 4)),
+            UnorientedEdge(OrientedEdge::new(4, 5)),
+            UnorientedEdge(OrientedEdge::new(5, 3)),
+            UnorientedEdge(OrientedEdge::new(3, 0)),
+        ];
+
+        let calculated_loops = border_edge_loops(&edge_sharing_map);
+
+        assert_eq!(calculated_loops.len(), 1);
+        for edge in correct_loop {
```

Comment:
Hmmm, can't we in this case assert also the order? In other words: `assert_eq!(calculated_loops[0], correct_loop);`?

Alternatively, let's at least also assert the length of the loop.

Reply:
you're right. the only problem is that I don't know which edge comes first.
let's check the length though